spec/predicates/release.md

```diff
@@ -0,0 +1,95 @@
+# Predicate type: Release
+
+Type URI: https://in-toto.io/attestation/release
+
+Version 0.1
+
+## Purpose
+
+To authoritatively link from a specific release name and version string in a registry, to the artifact names and hashes that make up that release.
+
+## Use Cases
+
+When receiving a new release version, package registries can publish a release attestation covering the artifact names and hashes that make up that release. This allows consumers of that release version to ensure the artifacts they are consuming have not been tampered with.
+
+If these release attestations are optionally published to a transparency log, package authors (or other interested parties) can monitor when a new version of a package is released.
+
+These use cases are not hypothetical; both of them are the case today for npm's [build provenance feature], which includes a [publish attestation]. Note that while npm calls this a publish attestation, calling it a release attestation better reflects that it's coming from the package registry. Publishing often refers to an author sending content to the registry, as in [PyPI's trusted publishers feature].
+
+## Prerequisites
+
+This predicate depends on the [in-toto attestation framework], as well as the [purl-spec] for identifying packaging ecosystems.
+
+Perhaps surprisingly, this predicate does not depend on [SLSA Provenance], but they are better together. In an ecosystem where some (but not all) packages have SLSA Provenance and you query by artifact hash, you can't tell the difference between an artifact that has been tampered with and one that does not yet have SLSA provenance. If the ecosystem has release attestations, you can give authoritative answers to what artifacts make up a given release version, and what hashes those artifacts should have.
+
+## Model
+
+This predicate is for the final stages of the software supply chain, where consumers are looking for attestations that the software has not been tampered with during distribution.
+
+## Schema
+
+### Fields
+
+- **purl, required** string (ResourceURI)
+  - A purl uniquely identifying a specific release name and version from a package registry.
+
+### Parsing Rules
+
+The purl field MUST be parsed using the [purl-spec]. It MUST include a version (which is OPTIONAL in the [purl-spec]).
+
+## Example
+
+```json
+{
+  "_type": "https://in-toto.io/Statement/v1",
+  "subject": [
+    {
+      "name": "@scope/my-package",
+      "digest": {
+        "sha256": "deadbeef0123456789..."
+      }
+    }
+  ],
+  "predicateType": "https://in-toto.io/attestation/release/v0.1",
+  "predicate": {
```

Comment:
Is there any metadata about the `authorized user` that could be added to the attestation? This could allow consumers to revoke trust for this attestation in the future, based on a policy.

Reply:
For the npm publish attestation (and npm build provenance) we deliberately omit account identity for several reasons:

- Strongly identifying and uniquely mapping humans to digital identities is a very hard problem at global scale
- Account behavior changes over time: either because people change or because their account is compromised
- Even if you had account identity here, how would generate (and distribute in a trustworthy way) what accounts should be trusted over what period of time, in order to write policy for a given release?

That's not to say that these aren't problems worth addressing in the future. But I feel strongly that they are out of scope of this iteration of release attestations.

Reply:
Based on the discussion we are going to leave out account identity.